code/game/mecha/mecha_wreckage.dm

```diff
@@ -59,6 +59,8 @@
 /obj/structure/mecha_wreckage/attackby(obj/item/I, mob/user, params)
 	if(!can_be_reconstructed)
 		return ..()
+	if(repair_efficiency <= 0)
```

Comment:
```suggestion
	if(!repair_efficiency)
```

Reply:
negative numbers are truthy, if this bites anyone in the ass in the future its on you